src/vs/workbench/contrib/chat/browser/chatContentParts/chatReferencesContentPart.ts

```diff
@@ -286,7 +286,8 @@ class CollapsibleListRenderer implements IListRenderer<IChatCollapsibleListItem,
 					}, { icon, title: data.options?.status?.description ?? data.title });
 			} else {
 				const variable = this.chatVariablesService.getVariable(reference.variableName);
-				templateData.label.setLabel(`#${reference.variableName}`, undefined, { title: data.options?.status?.description ?? variable?.description });
+				// This is a hack to get chat attachment ThemeIcons to render for resource labels
+				templateData.label.setLabel(variable?.fullName ? `${variable.icon ? `$(${variable.icon.id}) ` : ''}${variable?.fullName}` : `#${reference.variableName}`, `#${reference.variableName}`, { title: data.options?.status?.description ?? variable?.description });
```

Comment:
the nested ternary AND null coalescing. I love TS.

Reply:
Nested template string 🥵